### PR TITLE
DSD-1051: Updating and removing npm packages

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-1051](https://jira.nypl.org/browse/DSD-1051)

## This PR does the following:

- Update Storybook packages and removes others that are now defaults packages/settings in Storybook.

## How has this been tested?

Locally.

## Accessibility concerns or updates

- N/A

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: .storybook/main.js [hunk 1]
```diff
@@ -15,18 +15,7 @@ module.exports = {
...
     "@storybook/addon-essentials",
-    {
-      name: "@storybook/addon-docs",
-      options: {
-        configureJSX: true,
-        // This allows Storybook to read the doc style comments for
-        // components and their props.
-        sourceLoaderOptions: {
-          parser: "typescript",
-          injectStoryParameters: true,
```
Comment (EdwinGuzman) [inline .storybook/main.js:26]: These are all now part of `addon-docs`.

File: .storybook/preview.tsx [hunk 2]
```diff
@@ -1,20 +1,14 @@
...
 import { withTests } from "@storybook/addon-jest";
+import React from "react";
 
 import nyplTheme from "../src/theme";
 import results from "../.jest-test-results.json";
 
 addParameters({
-  docs: {
-    container: DocsContainer,
-    page: DocsPage,
```
Comment (EdwinGuzman) [inline .storybook/preview.tsx:16]: These are now defaults as well.

File: package.json [hunk 3]
```diff
@@ -80,32 +80,31 @@
   "devDependencies": {
     "@babel/core": "7.14.6",
     "@chakra-ui/storybook-addon": "1.0.3",
-    "@mdx-js/react": "1.6.22",
```
Comment (EdwinGuzman) [inline package.json:83]: This was actually not used before. There is a similar package used in Storybook internally.

File: package.json [hunk 4]
```diff
@@ -80,32 +80,31 @@
...
+    "@jest/core": "27.3.1",
```
Comment (EdwinGuzman) [inline package.json:83]: This is to fix a security issue.

**Review by bigfishdesign13 — APPROVED** [review]
Sterling.

**Pull request merged**